site/using-smart-accounts/session-keys/supported-permissions.md

```diff
@@ -0,0 +1,347 @@
+
+Alternatively, you can also require that a session key uses a specific paymaster address, instead of spending the account’s native token for gas.
+
+## Importance of Gas Limits
+
+Gas spend limits are critically important to protecting the account. If you are using a session key, you should configure either a required paymaster rule or a gas spend limit. Failing to do so could allow a compromised session key to drain the account’s native token balance.
+
+## Default values
+
+Permissions start with the following default values:
+
+| Permission               | Default Value                                                                                                                                    |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Access control list      | Type: allowlist <br /> The list starts empty. When the allowlist is empty, all calls will be denied.                                             |
+| Time range               | Unlimited                                                                                                                                        |
+| Native token spend limit | 0 <br /> This means all calls spending the native token will be denied, unless the limit is updated or removed.                                  |
+| ERC-20 spend limit       | Unset. If you want to enabled an ERC-20 spend limit, add the ERC-20 token contract to the access control list and set the spending limit amount. |
+| Gas spend limits         | Unset. When defining the session key’s permissions, you should specify either a spending limit or a required paymaster.                          |
+
+## Using the PermissionsBuilder
+
+To construct the data to set a key's permissions, you will need to use the `SessionKeyPermissionBuilder` class. This will allow you to specify a series of updates, and when complete, you may generate the encoded data to perform all updates at once.
+
+The permissions data may be specified in 3 places:
+
+- In the Session Key Plugin's `onInstall` data, setting the intial permissions for a session key added at install time.
+- As data for the initial permissions of a session key added via `addSessionKey`.
+- As a parameter to the `updateKeyPermissions` function, to change the permissions of an existing key.
+
+#### Generating the permissions
+
+```ts
+// Let's create an initial permission set for the session key giving it an eth spend limit
+const keyPermissions = new SessionKeyPermissionsBuilder()
+  .setNativeTokenSpendLimit({
+    spendLimit: 1000000n,
+  })
+  // this will allow the session key plugin to interact with all addresses
+  .setContractAccessControlType(SessionKeyAccessListType.ALLOW_ALL_ACCESS)
+  .setTimeRange({
+    validFrom: Math.round(Date.now() / 1000),
+    // valid for 1 hour
+    validUntil: Math.round(Date.now() / 1000 + 60 * 60),
+  });
+```
```

Comment:
nit: for this and the examples below, it's best to put these in `/snippets` so that you get intellisense feedback. At somepoint I'll also be running `tsc` over those files to make sure there's no errors when building which should catch errors and help keep docs in sync across code changes.

Reply:
done!